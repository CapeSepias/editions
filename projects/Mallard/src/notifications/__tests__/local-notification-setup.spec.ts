import {
    notificationId,
    notificationPayload,
} from '../local-notification-setup'
import moment from 'moment'
import { regionalEdition } from './__fixtures__/regional-edition'

describe('Local Notification Setup', () => {
    describe('notificationId', () => {
        it('should return the notification in a {YYYYMMDD} format', () => {
            const id = notificationId(
                'australian-edition',
                moment('2020-09-02T13:57:03.896Z'),
            )
            expect(id).toEqual('20200902')
        })
    })

    describe('notificationPayload', () => {
        it('should deliver a matching notification payload', () => {
            const payload = notificationPayload(
                regionalEdition,
                moment('2020-09-02T13:57:03.896Z'),
            )
            expect(payload).toMatchSnapshot()
        })
    })
})
