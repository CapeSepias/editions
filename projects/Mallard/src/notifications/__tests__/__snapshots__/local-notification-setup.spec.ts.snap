// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Local Notification Setup notificationPayload should deliver a matching notification payload 1`] = `
Object {
  "allowWhileIdle": true,
  "date": 2020-09-02T13:57:03.896Z,
  "id": "20200902",
  "message": "The latest edition of Australia Weekend is available",
  "title": "Australia Weekend",
}
`;
